File: app/src/main/java/aliabbas/com/codebaseapp/di_dependencies/app_dependencies/MainActivityRefMod.java
package aliabbas.com.codebaseapp.di_dependencies.app_dependencies;

import android.app.Activity;

import aliabbas.com.codebaseapp.activity.MainActivity;
import dagger.Binds;
import dagger.Module;

/*
 * Created by Ali Abbas
 * This module is used to tell the DI about the Network helpers we have and the object we need to inject.
 *
 */
@Module
public abstract class MainActivityRefMod {

    @Binds
    abstract Activity homeActivityRefrence(MainActivity homeActivity);


}

File: app/src/main/java/aliabbas/com/codebaseapp/di_dependencies/app_dependencies/MainActivityRefMod.kt
package aliabbas.com.codebaseapp.di_dependencies.app_dependencies

import aliabbas.com.codebaseapp.activity.MainActivity
import android.app.Activity
import dagger.Binds
import dagger.Module

/*
 * Created by Ali Abbas
 * This module is used to inject activity references
 *
 */
@Module
abstract class MainActivityRefMod {

    //With the help of this We can Inject the activity in Dependent classes
    //Making activity reference available for injection.
    @Binds
    abstract fun homeActivityRefrence(homeActivity: MainActivity): Activity
}